app/controllers/challenge_partnerships_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class ChallengePartnershipsController < ApplicationController
+  before_action :set_form, only: %i[show create]
+
+  def show; end
+
+  def create
+    render :show and return unless @challenge_partnership_form.valid?
+
+    @challenge_partnership_form.challenge!
```

Comment:
Ah, that is interesting! Basically you've converted the form object into a domain service object, which is responsible for both validation, data storage and business logic. I very much love this idea - this is my preferable way of structuring the project. :)